Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverMode.hs

```diff
@@ -40,5 +51,187 @@ setDriverMode apiKey req = do
+      incrementKeysWithCounts (map (fromMaybe 0) newFleetRedisCount)
+
+-- | Decrement the fleet/operator status key for a given entityId and status
+decrementFleetOperatorStatusKeyForDriver ::
+  ( MonadFlow m,
+    EsqDBFlow m r,
+    CacheFlow m r,
+    Redis.HedisFlow m r,
+    HasField "serviceClickhouseCfg" r CH.ClickhouseCfg,
+    HasField "serviceClickhouseEnv" r CH.ClickhouseEnv
+  ) =>
+  DP.Role ->
+  Text ->
+  Maybe DDFS.DriverFlowStatus ->
+  m ()
+decrementFleetOperatorStatusKeyForDriver entityType entityId mbStatus = do
+  let status = fromMaybe DDFS.INACTIVE mbStatus
+  let key = DDFS.getStatusKey entityId status
+  keyExists <- Redis.get @T.Text key
+  let msg = "Key does not exist for driver status key: " <> show key <> " for status=" <> show status <> ", entityId=" <> show entityId
+  if isNothing keyExists
+    then fork msg $ do
+      void $ SDF.handleCacheMissForDriverFlowStatus entityType entityId (DDFS.allKeys entityId)
+      void $ Redis.decr key
+    else do
+      logInfo $ "Key already exists for driver status key: " <> show key <> " for status=" <> show status <> ", entityId=" <> show entityId
+      void $ Redis.decr key
+
+decrementOperatorStatusKeyForFleetOwner ::
+  ( MonadFlow m,
+    EsqDBFlow m r,
+    CacheFlow m r,
+    Redis.HedisFlow m r,
+    HasField "serviceClickhouseCfg" r CH.ClickhouseCfg,
+    HasField "serviceClickhouseEnv" r CH.ClickhouseEnv
+  ) =>
+  Text ->
+  Text ->
+  m ()
+decrementOperatorStatusKeyForFleetOwner operatorId fleetOwnerId = do
+  let fleetOwnerKeys = DDFS.allKeys fleetOwnerId
+      operatorKeys = DDFS.allKeys operatorId
+
+  fleetRedisCount <- mapM (Redis.get @Int) fleetOwnerKeys
+  operatorRedisCount <- mapM (Redis.get @Int) operatorKeys
+
+  let decrementKeysWithCounts counts =
+        forM_ (zip operatorKeys counts) $ \(operatorKey, fleetCount) ->
+          Redis.decrby operatorKey (fromIntegral fleetCount)
+
+      handleFallback role entityId keyData = do
+        void $ SDF.handleCacheMissForDriverFlowStatus role entityId keyData
+        mapM (Redis.get @Int) keyData
+
+  case (all isJust fleetRedisCount, all isJust operatorRedisCount) of
+    (True, True) -> decrementKeysWithCounts (map (fromMaybe 0) fleetRedisCount)
+    (True, False) -> fork "Operator key data not found" $ do
+      _ <- handleFallback DP.OPERATOR operatorId operatorKeys
+      decrementKeysWithCounts (map (fromMaybe 0) fleetRedisCount)
+    (False, True) -> fork "Fleet owner key data not found" $ do
+      newFleetRedisCount <- handleFallback DP.FLEET_OWNER fleetOwnerId fleetOwnerKeys
+      decrementKeysWithCounts (map (fromMaybe 0) newFleetRedisCount)
+    (False, False) -> fork "Both fleet owner and operator key data not found" $ do
+      newFleetRedisCount <- handleFallback DP.FLEET_OWNER fleetOwnerId fleetOwnerKeys
+      _ <- handleFallback DP.OPERATOR operatorId operatorKeys
+      decrementKeysWithCounts (map (fromMaybe 0) newFleetRedisCount)
```

Comment:
Do we need increment/decrement after fetching data from clickhouse? I mean that clickhouse data already should be actual, right?